graphics/src/widget/text_input.rs

```diff
@@ -107,6 +108,34 @@ where
 
         let text = value.to_string();
 
+        let current_text_bounds = {
+            if text.is_empty() {
+                let placeholder_visual_width =
+                    self.measure_value(&placeholder.to_string(), size, font);
```

Comment:
Do we need to measure again here?

There is a `text_width` measurement in line 250 we may be able to reuse to position things properly.

Reply:
`text_width` is for user typed text, while this one is for `placeholder_text` width.

This is required to place the `placeholder` text at the center/left-aligned when `horizontal_alignment` is `Center`/`Right`. This will only be calculated when user input is empty.

<img width="409" alt="image" src="https://user-images.githubusercontent.com/11786283/83474995-ff217600-a4aa-11ea-8d60-4bca3a79d529.png">


I couldn't think of any other way to make placeholder center/left-aligned.

Reply:
`text_width` measures the placeholder if the value is empty:

https://github.com/hecrj/iced/blob/718c8fd393987e9c4e1c486e03ce31175b5f94a3/graphics/src/widget/text_input.rs#L250-L254

Reply:
My Bad. I will move this measurement to the top then. 😅